templates/CRM/UF/Form/Fields.tpl

```diff
@@ -93,12 +93,15 @@
           ( $form.formName neq 'Confirm' )  AND
           ( $form.formName neq 'ThankYou' ) }
             {include file="CRM/common/jcalendar.tpl" elementName=$profileFieldName}
-          {elseif ( $profileFieldName|substr:-5:5 eq '_date' ) }
+          {elseif $field.is_datetime_field && $action == 4}
```

Comment:
$action == 4 - ie we are looking at a page with the action view- should be the case for confirm screens